### PR TITLE
Consolidate font sizes and line heights

### PR DESCRIPTION
**Pull request by lyzadanger**
Part of https://github.com/hypothesis/client/issues/2165

This PR attempts to consolidate `font-size` and `line-height` rules for the annotator and the client. It does not attempt to solve design problems (for the most part) but is part of an effort to continue to organize our existing applied design so that it can be better evaluated.

This PR does improve some consistency with font sizes and line heights in some components, and those will be itemized below, in commits and in code comments. 

Commits are granular; I suggest reviewing them for more info.

The state of the SASS after these changes:

* Sidebar components may use one of four font mixins for establishing font size and line height. Two already existed: `font-normal` (13px/17px), `font-small` (11px/15px). This PR adds `font-big` (14px/20px) and `font-tiny` (10px/14px).
* No changes were made to `annotator` font sizes or line heights, but the values used in annotator SASS modules have been corraled in the `variables` modules for later evaluation.
* The `font-size` of 12px, which was present in a few places before these changes, has been removed as an option. In its former stead, font sizes have either been bumped up to `font-normal` (13px presently) or down to `font-small` (11px presently). This is in attempt to stop having 1px gradation in our font sizes and is one of the few design assertions this set of changes makes.
* `$base-font-size` was an illusion. It was set to `16px` — a noble idea — but only used in one place, where it was immediately overridden to `12px`.
* One unused SASS module and some unused rules that involve font sizes were removed as I spotted them.
* A bug with a huge font size for the tag editor for mobile devices was rectified.

### Not part of this work

* Font colors, weights, etc. This is just about font-sizes and line-heights.
* Making any prescriptions as to how typography design should work, with the exception of removing the `12px` font size from the mix. It’s known that our base font size is too small, and that our rhythms aren’t established, etc.
* Naming and organization of typography SASS patterns. Yep, we might want to pull them out of `variables`. Yep, we likely want to add color and other rules to the mixins as they’re developed. This is primarily an effort toward tidying and analysis.

Goal is to keep moving fast and making helpful improvements without trying to bite the whole problem off in one piece.

### CODE REVIEW

File: src/styles/sidebar/components/annotation-quote.scss [hunk 1]
```diff
@@ -9,13 +9,9 @@
 
   &__quote {
     @include var.font-normal;
-
     border-left: 3px solid var.$grey-3;
     color: var.$color-text-light;
-    font-family: sans-serif;
-    font-size: 12px;
     font-style: italic;
-    letter-spacing: 0.1px;
```
Comment (lyzadanger) [inline src/styles/sidebar/components/annotation-quote.scss:18]: Typographer Frederic Goudy infamously lamented that “Anyone who would letterspace lowercase would steal sheep”.

File: src/styles/sidebar/components/logged-out-message.scss [hunk 2]
```diff
@@ -25,7 +25,8 @@
 }
 
 .logged-out-message__logo-icon {
-  font-size: 30px;
+  width: 30px;
```
Comment (lyzadanger) [inline src/styles/sidebar/components/logged-out-message.scss:28]: This needn't/shouldn't be a font-related rule.

File: src/styles/sidebar/components/tag-editor.scss [hunk 3]
```diff
@@ -28,7 +27,7 @@
     border: 1px solid var.$grey-3;
     border-radius: 2px 0 0 2px;
     border-right-width: 0;
-    padding: 0 0.5em;
+    padding: 2px 0.5em;
```
Comment (lyzadanger) [inline src/styles/sidebar/components/tag-editor.scss:30]: This deserves to be revisited but was formerly achieved by applying a custom `line-height` here.

**Comment by robertknight** [comment]
The group name has moved a little closer to the username and is now more prominent on this branch. I think this is OK as a step on the path to consolidating our typography, but IMO it gives us more impetus to reconsider the layout (eg. by removing it from the annotation card if all the visible annotations are in the same group).

Master:

group-label-master

This branch:

group-label-branch

**Comment by lyzadanger** [comment]
> The group name has moved a little closer to the username and is now more prominent on this branch. I think this is OK as a step on the path to consolidating our typography, but IMO it gives us more impetus to reconsider the layout (eg. by removing it from the annotation card if all the visible annotations are in the same group).

I agree with this wholeheartedly. I want to flag up that we're likely to see a bit more temporary design discomfort over the next few weeks.

While this PR attempts to tame the font-size beast, it doesn't actually solve some of the typography shortcomings of the design, so we'll want to revisit...

**Pull request merged**